lightstep/recorder.py

```diff
@@ -1,21 +1,76 @@
-""" This code is used to report traces to LightStep servers.
-"""
+from .basictracer.recorder import SpanRecorder
+from .crouton import ttypes
+
+import util
+import logging
+import constants
+import pprint
+import sys
 
 import atexit
 import contextlib
 import jsonpickle
 import random
 from socket import error as socket_error
 import ssl
-import sys
 import threading
 import time
 import warnings
 
 from thrift import Thrift
+from . import constants, version as cruntime_version, util, connection as conn #???
 
-from .crouton import ttypes
-from . import constants, version as cruntime_version, util, connection as conn
+
+class Recorder(SpanRecorder):
+    """ SpanRecorder's job is record and report a BasicSpan.
+    """
+    def __init__(self, *args, **kwargs):
+        self.runtime = Runtime(*args, **kwargs)
+        self._runtime_guid = self.runtime._runtime.guid
+
+    def record_span(self, span):
+        self.runtime._add_span(span)
+
+    def flush(self):
+        self.runtime.flush()
+
+def _pretty_logs(logs):
+    return ''.join(['\n  ' + pprint.pformat(log) for log in logs])
+
+class LoggingRecorder(SpanRecorder):
+
+    """Logs all spans to console."""
+
+    def __init__(self, *args, **kwargs):
+        self._runtime_guid = util._generate_guid()
+
+    def record_span(self,span):
```

Comment:
I'd vote for either (1) just logging the `span` directly (as a `BasicSpan`), or (2) factoring out the code that converts a `BasicSpan` to a `ttypes.SpanRecord` and invoking it here.

That way we'll either have (1) a super short, simple function, or (2) a slightly more complex thing that at least lets us see exactly what's sent over the wire.